### PR TITLE
Sm/remote-tracking-no-config-file

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?

Don't use ConfigFile
Preserve file format, but in-memory uses `Map`
Remove unused public methods (this module is NOT exported from STL so it's not a public API anyway, no breaking changes to be concerned about)

### What issues does this PR fix or reference?
@W-14399468@

### CODE REVIEW

File: src/shared/remoteSourceTrackingService.ts [hunk 1]
```diff
@@ -604,15 +526,58 @@ function getDecodedKeyIfSourceMembersHas({
...
 }
+
+const getFilePath = (orgId: string): string => path.join('.sf', 'orgs', orgId, FILENAME);
+
+const readFileContents = async (filePath: string): Promise<Contents | Record<string, never>> => {
+  try {
+    const contents = await fs.promises.readFile(filePath, 'utf8');
+    return parseJsonMap<Contents>(contents, filePath);
+  } catch (e) {
+    return {};
```
Comment (shetzel) [inline src/shared/remoteSourceTrackingService.ts:551]: I think we should at least log the error, if not handle the errors and throw more meaningful ones.  Returning `{}` doesn't satisfy the return type does it?
When this function is called, the file is expected to be there and have parseable content so this shouldn't ever happen but if it does we should know the reason why, versus continue silently with empty state.

File: src/shared/remoteSourceTrackingService.ts [hunk 2]
```diff
@@ -545,31 +470,28 @@ export class RemoteSourceTrackingService extends ConfigFile<RemoteSourceTracking
...
-    }
-
-    try {
-      return (await this.org.getConnection().tooling.query<SourceMember>(query, { autoFetch: true, maxFetch: 50000 }))
-        .records;
-    } catch (error) {
-      throw SfError.wrap(error as Error);
-    }
+  private async write(): Promise<void> {
+    await fs.promises.mkdir(path.dirname(this.filePath), { recursive: true });
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/shared/remoteSourceTrackingService.ts:483]: I'd really like it not to throw ENOENT in the case where someone move/deletes the file, etc, as a safeguard.

Most of those should be no-ops, though.

File: src/shared/remoteSourceTrackingService.ts (hunk 1 above)
Comment (mshanemc) [reply src/shared/remoteSourceTrackingService.ts:551]: logged.  It's not necessarily an error if there is no file there yet to read.  

The type can be an empty {} (that's the Record<string, never> alternative)

File: src/shared/remoteSourceTrackingService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/shared/remoteSourceTrackingService.ts:482]: alternatively, we could write the initial file on init => read so write doesn't have to?

File: src/shared/remoteSourceTrackingService.ts (hunk 1 above)
Comment (mshanemc) [reply src/shared/remoteSourceTrackingService.ts:555]: alternatively, we could change it to catch the ENOENT and re-throw other things?

**Review by shetzel — APPROVED** [review]

**Comment by shetzel** [comment]
QA Notes (using dreamhouse-lwc project).  All actions below were as expected so a global ✅ 

1. create a scratch org and `project deploy preview`
2. `project retrieve preview`
3. `project deploy start`
4. `project deploy preview`
5. make remote change and `project retrieve preview`
6. `project retrieve start`
7. make local change and `project deploy preview`
8. `project deploy start`
9. make local and remote change to same metadata and `project deploy preview` and `project retrieve preview`
10. `project deploy start --ignore-conflicts`
11. `project deploy preview` and `project retrieve preview`

create a new project (sf project generate), auth to the scratch org used above, then
12. `project retrieve preview`
13. `project retrieve start` pulls all the source
14. `project retrieve preview`

**Comment by shetzel** [comment]
Also ran
`project reset tracking` and `project delete tracking` and both behaved as expected.

**Pull request merged**